### ISSUE TITLE
PDF support

### ISSUE THREAD

**Issue by aartaka** [atlas-engineer/nyxt#2164]
**Is your feature request related to a problem? Please describe.**
Lots of browsers have built-in PDF-viewer (even if it's just a barebones [mozilla/pdf.js](https://github.com/mozilla/pdf.js)), and that's a valid expectation that Nyxt, as a browser, has it too. 

**Describe the solution you'd like**
Add a Nyxt-native PDF-reader, most likely as a custom scheme (Epiphany does that) embedding pdf.js reader.

**Describe alternatives you've considered**
- Add PDF-specific hooks and expand documentation so that one can easily configure their favorite PDF reader to open once a PDF is being downloaded.
- Hope for an extension with PDF support.
  - However, extensions are quite an indirect way to do thing, and PDF support is better as a built-in feature. 

**Additional context**
Epiphany does it this way:
- Add a custom `ephy-pdf:` scheme.
- When the content type of the link is PDF, redirect to `ephy-pdf:` page.
- When on `ephy-pdf:`, load a slightly customized [pdf.js template viewer](https://github.com/mozilla/pdf.js/blob/master/web/viewer.html) with a reference to the PDF being loaded.
- Et voila! pdf.js renders the PDF and allows interacting with it conveniently.

**Comment by Ambrevar** [comment]
On a related note, there is pure CL library to generate PDF: https://github.com/mbattyani/cl-pdf

**Comment by hendursaga** [comment]
Stop the presses!

The [new stable WebKitGTK release](https://webkitgtk.org/2022/09/16/webkitgtk2.38.0-released.html) supports PDF documents out-of-the-box using PDF.js!

**Comment by aartaka** [comment]
Whaaaaat! No way...

**Comment by jmercouris** [comment]
That's incredible!

**Comment by shaunsingh** [comment]
Considering gnome web on webkitgtk supports pdf's via pdf.js on webkitgtk 2.38 and onwards, I think this should be doable within nyxt now. 

What would be needed to implement pdf support in nyxt? I tried looking through the source to see how nyxt handles loading/displaying images, but I didn't find anything. 

For what its worth, the pdf.js demo works just fine in nyxt, but I think implementing vim bindings ala zathura would be nice

**Comment by jmercouris** [comment]
I don't believe we'll need to do anything other than wait to package Nyxt with WebKitGTK 2.38+

**Comment by aadcg** [comment]
Does #2988 change the status of this issue?
